src/cpp/dds/domain/DomainParticipant.cpp

```diff
@@ -128,10 +128,6 @@ void DomainParticipant::default_participant_qos(
     {
         throw dds::core::InconsistentPolicyError("Inconsistent Qos");
     }
-    else if (code == ReturnCode_t::RETCODE_IMMUTABLE_POLICY)
-    {
-        throw dds::core::ImmutablePolicyError("Immutable Qos");
-    }
```

Comment:
Add also a check UNSUPPORTED. Although it may not be returned at present, it covers future possibilities.